### DISCUSSION TITLE
'ready' event not fired when using pre-decoded peaks data

### DISCUSSION THREAD

**Discussion by jahangiranwari** [katspaugh/wavesurfer.js#3287] [Q&A] [answered]
Following PR #3283 the expectation would be that when we provide pre-decoded peaks data and duration to in options to `WaveSurfer.create` the wavesurfer instance would behave the same way as `load("", peaks, duration)`. However, on testing this feature we noticed that when providing pre-decoded peaks data and duration as `WaveSurferOptions` the waveform is rendered but the `ready` event is not fired/emitted. 

On looking through the code it appears that in the old way when using `load` it would call `loadAudio` which at the end would emit [ready](https://github.com/katspaugh/wavesurfer.js/blob/ee46b3222fa4ed4171c2e3d64d5588fbc3298b42/src/wavesurfer.ts#L382) event. But when using pre-decoded peaks data we call [loadPredecoded](https://github.com/katspaugh/wavesurfer.js/blob/ee46b3222fa4ed4171c2e3d64d5588fbc3298b42/src/wavesurfer.ts#L332) but unfortunately this method doesn't emit the `ready` event before returning.

**Comment by katspaugh** [comment, marked as answer]
Ready is fired when an instance is rendered and ready for playback.

Without audio it will only emit the decode and redraw events, and this is by design.

**Reply by jahangiranwari** [reply]
Ahh okay. We misunderstood the `ready` event. With the new feature to render with pre-decoded data we assumed we could replace below:
```javascript
const options = { container: "#waveform" }
const ws = WaveSurfer.create(options)
ws.on("ready", () => {
  // perform some action
})
await ws.load("", peaks, duration)
```
with:
```javascript
const options = { container: "#waveform", peaks: peaks, duration: duration }
const ws = WaveSurfer.create(options)
ws.on("ready", () => {
  // perform some action
})

```

**Reply by katspaugh** [reply]
Gotcha. You can probably use the `decode` event for this, just be aware that it will fire again if you call `load` with an auido URL subsequently.
